controllers/topology/controller.go

```diff
@@ -141,6 +141,20 @@ func (r *ClusterReconciler) reconcile(ctx context.Context, s *scope.Scope) (ctrl
 		return ctrl.Result{}, errors.Wrap(err, "error reading current state of the Cluster topology")
 	}
 
+	// Watch Infrastructure and ControlPlane CRs when they exists.
+	if s.Current.InfrastructureCluster != nil {
+		if err := r.externalTracker.Watch(ctrl.LoggerFrom(ctx), s.Current.InfrastructureCluster,
+			&handler.EnqueueRequestForOwner{OwnerType: &clusterv1.Cluster{}}); err != nil {
+			return ctrl.Result{}, err
```

Comment:
Should we wrap this and the other error with additional information so that we are consistent with other error returns in this function?